src/main/webapp/MyMarker.js

```diff
@@ -12,36 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-class Marker {
+class MyMarker {
```

Comment:
nit: Usually "My" is used as a prefix for instances, not classes. Maybe you can use something like PermanentMarker (since you have TempMarker) or ChapMarker. Same thing goes to MyMap.